Sources/Misc/RawDataContainer.swift

```diff
@@ -11,14 +11,33 @@
 //
 //  Created by Nacho Soto on 11/16/21.
 
-/// A type which exposes its underlying raw data, for debugging purposes or for getting access
+/// A type which exposes its underlying content for debugging purposes or for getting access
 /// to future data while using an older version of the SDK.
 public protocol RawDataContainer {
 
-    /// The type of the underlying raw data
-    associatedtype Content
+    /// The type of the `underlyingData` for this type.
+    associatedtype Content: Encodable
 
-    /// The underlying data.
-    var rawData: Content { get }
+    /// The underlying content for debugging purposes or for getting access
+    /// to future data while using an older version of the SDK.
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+    var underlyingData: Content { get }
+
+}
+
+/// Default implementation that encodes the `underlyingData` for public use.
+extension RawDataContainer {
+
+    /// The underlying data for this type, for debugging purposes or for getting access
+    /// to future data while using an older version of the SDK.
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+    public var rawData: [String: Any] {
+        do {
+            return try self.underlyingData.asDictionary()
+        } catch {
+            Logger.warn(Strings.codable.encoding_error(error))
+            return [:]
+        }
+    }
```

Comment:
Will this `rawData` not contain any data that was stripped away by the new lossy array and dictionary types? So like.. this won't show the raw response that this customer info was created from, right? It instead is just a dictionary representation of this object?

I'm not sure if that is a change from what this previously did but just wanted to understand and call that out just in case 😇

Reply:
Oh shoot that's a great catch! Let me look into how we could support storing the original data.

Reply:
Forgot to update this here, this is fixed by #1565.